src/Atmos/Parameterizations/CloudPhysics/Microphysics.jl

```diff
@@ -1,271 +1,575 @@
+                                 q::PhasePartition{FT}, q_rai::FT, ρ::FT,
+                                 T::FT) where {FT<:Real}
+    evap_subl_rate = FT(0)
+    if q_rai  > FT(0)
+
+        _a_vent::FT  = a_vent(rain_param_set)
+        _b_vent::FT  = b_vent(rain_param_set)
+        _ν_air::FT   = ν_air(param_set)
+        _D_vapor::FT = D_vapor(param_set)
+
+        _S::FT = supersaturation(param_set, q, ρ, T, Liquid())
+        _G::FT = G_func(param_set, T, Liquid())
+
+        (_n0, _α, _β, _γ, _δ, _ζ, _η) =
+            unpack_params(param_set, rain_param_set, ρ, q_rai)
+        _λ::FT = lambda(q_rai, ρ, _n0, _α, _β)
+
+        evap_subl_rate = 4 * π * _n0 / ρ * _S * _G / _λ^FT(2) * (
+          _a_vent +
+          _b_vent * (_ν_air / _D_vapor)^FT(1/3) * (FT(2) * _ζ / _ν_air)^FT(1/2) *
+            gamma((_η + FT(5)) / FT(2)) / _λ^((_η + FT(1))/FT(2))
+        )
+    end
+    return evap_subl_rate
+end
+function evaporation_sublimation(param_set::APS, snow_param_set::ASPS,
+                                 q::PhasePartition{FT}, q_sno::FT, ρ::FT,
+                                 T::FT) where {FT<:Real}
+    evap_subl_rate = FT(0)
+    if q_sno  > FT(0)
+
+        _a_vent::FT  = a_vent(snow_param_set)
+        _b_vent::FT  = b_vent(snow_param_set)
+        _ν_air::FT   = ν_air(param_set)
+        _D_vapor::FT = D_vapor(param_set)
+
+        _S::FT = supersaturation(param_set, q, ρ, T, Ice())
+        _G::FT = G_func(param_set, T, Ice())
+
+        (_n0, _α, _β, _γ, _δ, _ζ, _η) =
+            unpack_params(param_set, snow_param_set, ρ, q_sno)
+
+        _λ::FT = lambda(q_sno, ρ, _n0, _α, _β)
+
+        evap_subl_rate = 4 * π * _n0 / ρ * _S * _G / _λ^FT(2) * (
+          _a_vent +
+          _b_vent * (_ν_air / _D_vapor)^FT(1/3) * (FT(2) * _ζ / _ν_air)^FT(1/2) *
+            gamma((_η + FT(5)) / FT(2)) / _λ^((_η + FT(1))/FT(2))
+        )
+    end
+    return evap_subl_rate
+end
+
+"""
+    snow_melt(param_set, snow_param_set, q_sno, ρ)
+
+ - `param_set` - abstract set with earth parameters
+ - `snow_param_set` - abstract set with snow microphysics parameters
+ - `q_sno` - snow water specific humidity
+ - `ρ` - air density
+
+Returns the tendency due to snow melt.
+"""
+function snow_melt(param_set::APS, snow_param_set::ASPS, q_sno::FT,
+                   ρ::FT) where {FT<:Real}
+
+    snow_melt_rate = FT(0)
+    if q_sno  > FT(0)
+
+        _a_vent::FT  = a_vent(snow_param_set)
+        _b_vent::FT  = b_vent(snow_param_set)
+        _ν_air::FT   = ν_air(param_et)
+        _D_vapor::FT = D_vapor(param_set)
+        _K_therm::FT = K_therm(param_set)
+
+        _T_freeze = T_freeze(param_set)
+        L = latent_heat_fusion(param_set, T)
+
+        (_n0, _α, _β, _γ, _δ, _ζ, _η) =
+            unpack_params(param_set, snow_param_set, ρ, q)
+
+        _λ::FT = lambda(q_sno, ρ, _n0, _α, _β)
+
+        snow_melt_rate = 4 * π * _n0 / ρ *
+          _K_therm / L * (T - _T_freeze) / _λ^FT(2) * (
```

Comment:
Do you want a Heaviside function on `(T - _T_freeze)` here?

Reply:
done